data-service/src/test/java/com/amazonaws/blox/dataservice/integration/CreateEnvironmentIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.integration;
+
+import static org.junit.Assert.assertEquals;
+
+import com.amazonaws.blox.dataservicemodel.v1.exception.ResourceExistsException;
+import com.amazonaws.blox.dataservicemodel.v1.model.EnvironmentId;
+import com.amazonaws.blox.dataservicemodel.v1.model.wrappers.CreateEnvironmentResponse;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class CreateEnvironmentIntegrationTest extends DataServiceIntegrationTestBase {
+  private static final String ACCOUNT_ID = "123456789012";
+  private static final String ENVIRONMENT_NAME = "environmentName";
+  private static final String CLUSTER_ONE = "cluster1";
+  private static final String CLUSTER_TWO = "cluster2";
+  private static final String TASK_DEFINITION = "taskDefinition";
+
+  DataServiceModelBuilder models = DataServiceModelBuilder.builder().build();
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testCreateEnvironmentSuccessful() throws Exception {
+    final EnvironmentId createdEnvironmentId =
+        models
+            .environmentId()
+            .accountId(ACCOUNT_ID)
+            .environmentName(ENVIRONMENT_NAME)
+            .cluster(CLUSTER_ONE)
+            .build();
+    // I create an environment
+    final CreateEnvironmentResponse createEnvironmentResponse =
+        dataService.createEnvironment(
+            models
+                .createEnvironmentRequest()
+                .taskDefinition(TASK_DEFINITION)
+                .environmentId(createdEnvironmentId)
+                .build());
+
+    // The environment is valid
+    checkEnvironmentValid(ENVIRONMENT_NAME, ACCOUNT_ID, CLUSTER_ONE, createEnvironmentResponse);
+  }
+
+  @Test
+  public void testCreateAnEnvironmentAlreadyExist() throws Exception {
+    final EnvironmentId createdEnvironmentId =
+        models
+            .environmentId()
```

Comment:
EnvironmentId is the same everywhere and not really important to have it for each test case.  How about moving it as field?

Reply:
It's only on the third test that the id is different (having two different clusters). Do you suggest just create these two IDs as fields?

Reply:
y, could have a default environment ID. As most of the test case don't care it too much, the creation of environment ID (7 lines of code) is quite distraction to read the test.